### PR TITLE
migration to dune

### PR DESCRIPTION
**Pull request by c-cube**

### CODE REVIEW

File: tests/jbuild [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   (deps (test_toml.exe))
+   (action
+     (progn
+       (setenv BISECT_FILE coverage
+         (run ./test_toml.exe))
+     (run sh -c "bisect-report -I .*.eobjs -verbose -html report coverage*.out")
+     ))))
+;(executable
+;  ((name parser_with_coverage)
+;   (libraries (ounit toml bisect))))
```
Comment (mackwic) [inline tests/jbuild]: is this code dead ?

In this case: why keep it ?

File: toml-cconv.opam [hunk 2]
```diff
@@ -1,28 +1,17 @@
 opam-version: "1.2"
-
-available: [ ocaml-version >= "4.01.0" ]
-
+available: [ ocaml-version >= "4.02.0" ]
 version: "4.0.0"
```
Comment (mackwic) [inline toml-cconv.opam]: I think we'll need a version bump. Maybe not in this PR

**Comment by c-cube** [comment]
[dune](https://github.com/ocaml/dune) is the new build system that finally did what no one thought was possible: more or less become the de facto standard build system for OCaml. The normal build should be the same (`make`, `make test`, `make clean`?) and most people use opam anyway.

File: tests/jbuild (hunk 1 above)
Comment (c-cube) [reply tests/jbuild]: I'm not sure what it's supposed to do/output. I haven't used bisect in a while…

Comment (mackwic) [reply tests/jbuild]: if it's dead code, delete it. It's simpler

**Comment by mackwic** [comment]
Could you update the readme with the correct infos ? I don't know how to use `dune`.

Particularly, this is now outdated:
```mk
README.md:
  43   │ * From source:
  44   │ ```bash
  45   │ git clone https://github.com/sagotch/To.ml
  46   │ cd To.ml
  47   │ make build
  48   │ make install
  49   │ ```
```

**Comment by sagotch** [comment]
I dit not try to compile, but :+1: 

`src/cconv/jbuild` should be `src/cconv/jbuild`

If everything works fine, I :+1: the build system swtich.

**Comment by mackwic** [comment]
If it can help, here are the results of the different make rules:
- [x] `make build` OK
- [ ] `make build-cconv` KO (`Error: Library "cconv" not found`)
- [ ] `make install` KO ( `_build/default/toml-cconv.install is missing`)
- [ ] `make install-cconv` KO (`Error: Library "cconv" not found.`)
- [ ] `make test` KO (`Library "oUnit" not found`)
- [ ] `make doc` KO (`Error: Program odoc not found in the tree or in PATH (context: default)`)
- [x] `make clean` OK

**Comment by c-cube** [comment]
Ah, right, there are some assumptions (as with `_oasis`' generated `configure`, I suppose). They're all in the `opam` file.

- doc requires `odoc`
- `cconv` stuff requires `cconv` (…)

**Comment by mackwic** [comment]
build and install works, but doc, test, and the cconv version are still KO.

Is there a one line command we can use to tell dune to install the missing dependencies ?

**Comment by c-cube** [comment]
Yes, in fact it's already suggested in the error message:

$ make build-cconv
dune build toml-cconv.install
File "src/cconv/dune", line 5, characters 18-23:
Error: Library "cconv" not found.
Hint: try: dune external-lib-deps --missing toml-cconv.install
make: *** [Makefile:5: build-cconv] Error 1


"dune external-lib-deps --missing toml-cconv.install"

gives you:

$ dune external-lib-deps --missing toml-cconv.install
Error: The following libraries are missing in the default context:
- cconv
Hint: try: opam install cconv

So you can `opam install cconv`

**Comment by mackwic** [comment]
I, sorry I wasn't clear enough. I was asking if we needed to manually maintain the rule `make install-dependencies` as it's already specified in the .opam package.

Similar to python's `pip install -r requirements.txt`, to ruby's `bundle install`, or node's `npm install`, I was wondering if there was a standard way to install the dependencies of the package.

**Comment by c-cube** [comment]
Should I merge, then?

**Comment by mackwic** [comment]
yes ! :) 👍

**Pull request merged**